server/resp/src/main/java/org/infinispan/server/resp/Resp3AuthHandler.java

```diff
@@ -17,7 +18,11 @@
 public class Resp3AuthHandler extends CacheRespRequestHandler {
 
    public Resp3AuthHandler(RespServer server) {
-      super(server);
+      this(server, server.getCache());
+   }
+
+   public Resp3AuthHandler(RespServer server, AdvancedCache<byte[], byte[]> cache) {
```

Comment:
This constructor should be removed ino. It should only allow the RespServer single argument one and use the server cache, otherwise it could lead to passing the wrong cache.

Reply:
Argh, I was changing this but realized the Resp3Handler is a subclass of Resp3AuthHandler. I'll change the visibility to protected.